core/src/main/java/io/fundrequest/core/request/claim/github/GithubClaimResolver.java

```diff
@@ -56,44 +63,54 @@ public SignedClaim getSignedClaim(Principal user, UserClaimRequest userClaimRequ
-        return solver.isPresent() && solver.get().equalsIgnoreCase(userPlatformUsername);
+        final String owner = request.getIssueInformation().getOwner();
+        final String repo = request.getIssueInformation().getRepo();
+        final String number = request.getIssueInformation().getNumber();
+        return isIssueClosed(owner, repo, number) && getSolver(owner, repo, number).map(solver -> isClaimalbeByUser(user, request, solver)).orElse(false);
     }
 
-    public UserClaimableDto userClaimableResult(final Principal user, RequestDto request) {
-        final Optional<String> solver = getSolver(request);
-        if (solver.isPresent() && (request.getStatus() == RequestStatus.FUNDED || request.getStatus() == RequestStatus.CLAIMABLE)) {
-            return UserClaimableDto.builder()
-                                   .claimable(true)
-                                   .claimableByUser(isClaimalbeByUser(user, request, solver.get()))
-                                   .build();
+    public UserClaimableDto userClaimableResult(final Principal user, final RequestDto request) {
+        final IssueInformationDto issueInformation = request.getIssueInformation();
+        if (isIssueClosed(issueInformation.getOwner(), issueInformation.getRepo(), issueInformation.getNumber())) {
+            final Optional<String> solver = getSolver(request.getIssueInformation().getOwner(), request.getIssueInformation().getRepo(), request.getIssueInformation().getNumber());
+            if (solver.isPresent() && (request.getStatus() == RequestStatus.FUNDED || request.getStatus() == RequestStatus.CLAIMABLE)) {
+                return UserClaimableDto.builder()
+                                       .claimable(true)
+                                       .claimableByUser(isClaimalbeByUser(user, request, solver.get()))
+                                       .build();
+            }
         }
         return UserClaimableDto.builder().claimable(false).claimableByUser(false).build();
     }
 
-    @NotNull
-    private Boolean isClaimalbeByUser(Principal user, RequestDto request, String solver) {
+    private Boolean isClaimalbeByUser(final Principal user, final RequestDto request, final String solver) {
         return user == null
                ? false
                : getUserPlatformUsername(user, request.getIssueInformation().getPlatform()).map(u -> u.equalsIgnoreCase(solver)).orElse(false);
     }
 
-    private Optional<String> getSolver(final RequestDto request) {
-        final IssueInformationDto issueInformation = request.getIssueInformation();
-        return githubSolverResolver.resolveSolver(issueInformation.getOwner(), issueInformation.getRepo(), issueInformation.getNumber());
+    private boolean isIssueClosed(final String owner, final String repo, final String number) {
+        githubGateway.evictIssue(owner, repo, number);
```

Comment:
This will exhaust our api limit too fast, when you pull the solver from github, you can pull the issue status at the same time